### DISCUSSION TITLE
tasks.Wait make default retries of 180 seconds regardless of timedelta

### DISCUSSION THREAD

**Discussion by dannwise** [codingjoe/joeflow#80] [Q&A]
Hey @codingjoe, congratulations on the library. Very nice work.

Could you tell me if there is a way to configure the task retry time?

While using it I've noticed that whenever a machine task returns False it will auto-retry within 180 seconds (for celery). Even if I make the flow goes with a `tasks.Wait(timedelta(days=1))` for instance, the Wait task will return False and keep retrying every 180 seconds. So, tasks that should hang for long periods keep getting evaluated without need.

**Comment by codingjoe** [comment]
Hi @dannwise,

Thanks for your kind words. It means a lot <3
Are you using Celery or Dramatiq? Based on your choices, the back off behavior might vary.

This library doesn't implement its own back-off behavior, but relies on the default behavior of your task library. In both cases, the default behavior is well documented. A quick Google search should answer all your questions. If it doesn't, I am happy to help.

Cheers!
Joe

**Reply by danilNil** [reply]
Hi @codingjoe !

We faced with same question and looks like we need to rewrite [celery.py](https://github.com/codingjoe/joeflow/blob/main/joeflow/runner/celery.py) with proper interface for setting up parameter for auto retry. default_retry_delay in Celery is configured in [task decorator](https://docs.celeryq.dev/en/stable/reference/celery.app.task.html#celery.app.task.Task.default_retry_delay)

**Reply by codingjoe** [reply]
Is there any reason, why such a precise waiting time is needed? I, personally, didn't experience the use case yet, but I'd be happy to improve the implementation if need be.

**Reply by danilNil** [reply]
It's no longer relevant because we moved to Dramatiq, but it's relevant for Celery because it slows down debugging and development. We will be happy to contribute to the code as well if we will need that for dramatiq. But for now, it's ok since Dramatiq has a quicker default retrying period.

**Reply by codingjoe** [reply]
Hm… I am not so sure. I designed the task runner to be swappable on purpose. I think sticking to the default might best, since it reduces complexity and potential cause for error.
